### PR TITLE
add scheduler task that removes job log entries from the queue

### PR DESCRIPTION
**Pull request by mikethebeer**

### CODE REVIEW

File: sql/src/main/java/io/crate/operation/collect/StatsTables.java [hunk 1]
```diff
@@ -285,6 +297,15 @@ private void setJobsLog(int size, TimeValue expiration) {
         jobsLog.set(newQ);
     }
 
+    private class ScheduledQueueCleaner implements Runnable {
+
+        @Override
+        public void run() {
+            removeExpiredLogs((ConcurrentLinkedQueue<JobContextLog>) jobsLog.get(), System.currentTimeMillis(), lastJobsLogExpiration.getMillis());
+            LOGGER.info("removed expired jobs from queue");
```
Comment (mikethebeer) [inline sql/src/main/java/io/crate/operation/collect/StatsTables.java]: set to `LOGGER.debug` would be enough

Comment (chaudum) [inline sql/src/main/java/io/crate/operation/collect/StatsTables.java]: I don't think we need to log at all here

**Comment by mfussenegger** [comment]
Would it be possible to depend on a regular ScheduledExecutorService instead of the ThreadPool?

File: sql/src/main/java/io/crate/operation/collect/StatsTables.java [hunk 2]
```diff
@@ -263,18 +275,26 @@ private void setOperationsLog(int size) {
 
     private void setJobsLog(int size, TimeValue expiration) {
         Queue newQ;
+        boolean isScheduledQueue = false;
+
+        // Cancel the current jobs log queue scheduler if there is one
+        if (scheduledFuture != null) {
+            scheduledFuture.cancel(true);
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/operation/collect/StatsTables.java]: isn't it necessary to null it afterwards?

File: sql/src/main/java/io/crate/operation/collect/StatsTables.java [hunk 3]
```diff
@@ -64,7 +63,7 @@
 
     private final static BlockingQueue<OperationContextLog> NOOP_OPERATIONS_LOG = NoopQueue.instance();
     private final static BlockingQueue<JobContextLog> NOOP_JOBS_LOG = NoopQueue.instance();
-    private final static TimeValue QUEUE_CLEAN_INTERVAL = TimeValue.timeValueSeconds(5);
+    private final static long QUEUE_CLEAN_INTERVAL = 5L;
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/operation/collect/StatsTables.java]: please add a coment indicating the unit

**Pull request merged**